06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js

```diff
@@ -0,0 +1,186 @@
+  /**
+   * Creates an instance of InvertedIndex.
+   * @constructor
+   */
+  constructor() {
+    this.index = {};
+    this.temp_search = [];
+    this.searchResult = {};
+    this.allIndex = {};
+  }
+/**
+     * It returns the words passed in small letters and spaces removed
+     * @method tokenize
+     * @param {string} words
+     * @return {String}
+     */
+  tokenize(words) {
+    const token = words.replace(/,+/g, ' ')
+    .replace(/[^a-zA-Z 0-9\s]+/g, '')
+    .replace(/\s\s/g, ' ')
+    .toLowerCase()
+    .trim();
+    return token;
+  }
+  /**
+   * Reads the data from an uploaded file.
+   * @param {File} file - The file to be read
+   * @param {Function} callback - The callback function on file read
+   * @returns {void}
+   */
+  readFile(file, callback) {
+    const reader = new FileReader();
+    reader.onload = callback;
+    reader.readAsText(file);
+  }
+/**
+   * It validates the file passed and returns the error message if an error occurs
+   *
+   * @method validateFile
+   * @param {String} file
+   *
+   * @return {object} validates the file passed
+   */
+  validateFile(file) {
+    const jsonFile = file;
+    let check = {
+      status: true,
+      msg: 'Valid File',
+    };
+    try {
+      if (typeof file !== 'object' || file.length < 1) {
+        check = {
+          status: false,
+          msg: 'File is empty please upload a new file',
+        };
+      }
+      jsonFile.forEach((key) => {
+        if (key.title === undefined || key.text === undefined) {
+          check = {
+            status: false,
+            msg: 'Invalid file content',
+          };
+        }
+      });
+    } catch (error) {
+      return error.msg;
+    }
+  }
+/**
+     * It returns the index of the file passed
+     * @method createIndex
+     * @param {Object, Object} fileName jsonObject
+     * @return {number}
+     */
+  createIndex(fileName, jsonObject) {
+    const newIndex = {};
+    this.validateFile(jsonObject);
+
+    jsonObject.forEach((object, position) => {
+      const longSentence = `${object.title} ${object.text}`;
+      const tokenized = this.tokenize(longSentence);
+      const wordArray = tokenized.split(' ');
+
+      wordArray.forEach((word, pos) => {
+        if (newIndex[word] === undefined) {
+          newIndex[word] = [position];
+        } else if (newIndex[word].indexOf(position) < 0) {
+          newIndex[word].push(position);
+        }
+      });
+    });
+    this.index[fileName] = newIndex;
+  }
+/**
+     * It returns the index of the words
+     * @method getIndex
+     * @param {Object} fileName
+     *
+     * @return {object} gets the index
+     */
+  getIndex(fileName) {
+    if (Object.keys(arguments).length < 1) {
+      return this.index;
+    }
+
+    return this.index[fileName];
+  }
+  /**
+   *
+   *
+   * @method flattenSearch
+   * @param {void}
+   *
+   * @return {void}
+   */
+  flattenSearch() {
+    this.temp_search = [];
+    for (const arg of arguments) {
```

Comment:
Is this the correct use of the _**arguments**_ variable

Using 'ForOfStatement' is not allowed. Please see link your Airbnb style guide